statsmodels/multivariate/multivariate_ols.py

```diff
@@ -0,0 +1,472 @@
+
+    def __getitem__(self, item):
+        return self.results[item]
+
+    def f_test(self, hypothesis=None):
+        """
+        Testing the linear hypothesis
+            L * params * M = 0
+        where `params` is the regression coefficient matrix for the
+        linear model y = x * params, `M` is a dependent variable transform
+        matrix.
+
+        Parameters
+        ----------
+        hypothesis: A list of array-like
+            Hypothesis to be tested. Each element is an array-like
+                                [name, contrast_L, transform_M]
+            containing a string `name`, the contrast matrix L and the transform
+            matrix M (for transforming dependent variables), respectively.
+            contrast_L : array-like or an array of strings
+               Contrast matrix for hypothesis testing.
+               If array-like, each row is an hypothesis and each column is an
+               independent variable. At least 1 row
+               (1 by k_exog, the number of independent variables) is required.
+               If an array of strings, it will be passed to
+               patsy.DesignInfo().linear_constraint.
+            transform_M : array-like or an array of strings
+                Transform matrix.
+                Default to be k_endog by k_endog identity matrix
+                (i.e. do not transform y matrix).
+               If an array of strings, it will be passed to
+               patsy.DesignInfo().linear_constraint.
+            If `hypothesis` is None: 1) the effect of each independent variable
+            on the dependent variables will be tested. Or 2) if model is
+            created using a formula,  `hypothesis` will be created according to
+            `design_info`. 1) and 2) is equivalent if no additional variables
+            are created by the formula (e.g. dummy variables for categorical
+            variables and interaction terms)
+
+        Returns
+        -------
+        results: _MultivariateOLSResults
+
+        """
+        k_xvar = len(self.exog_names)
+        k_yvar = len(self.endog_names)
+        if hypothesis is None:
+            if self.design_info is not None:
+                terms = self.design_info.term_name_slices
+                hypothesis = []
+                for key in terms:
+                    L_contrast = np.eye(k_xvar)[terms[key], :]
+                    hypothesis.append([key, L_contrast, None])
+            else:
+                hypothesis = []
+                for i in range(k_xvar):
+                    name = 'x%d' % (i)
+                    L = np.zeros([1, k_xvar])
+                    L[i] = 1
+                    hypothesis.append([name, L, None])
+
+        results = _multivariate_ols_htest(hypothesis, self.fittedmod,
+                                          self.exog_names, self.endog_names)
+
+        self.results = results
+
+    def summary(self, contrast_L=False, transform_M=False):
```

Comment:
this is a summary for a test results, ContrastResults class
The standard summary for the ModelResults class has mainly the parameter estimates 

We can leave the `_MultivariateOLSResults.summary` method out for now, because we still need extra result like tvalues and pvalues, but I would like to move to a generic solution instead of adding another model specific version.